src/lib/bootloader/grub2pwd.rb

```diff
@@ -1,63 +1,131 @@
 require "yast"
 
-# class is responsible for detection, encryption and writing of grub2 password protection
-class GRUB2Pwd
-  YAST_BASH_PATH = Yast::Path.new(".target.bash_output")
-  PWD_ENCRYPTION_FILE = "/etc/grub.d/42_password"
+Yast.import "Stage"
 
-  def used?
-    Yast.import "FileUtils"
+module Bootloader
+  # class is responsible for detection, encryption and writing of grub2 password protection
+  class GRUB2Pwd
+    attr_accessor :used, :unrestricted
```

Comment:
Please document that they are Boolean.
Password being "used" is IMHO obvious but "unrestricted" is not, and this is a good place to document what we mean by it.